### PR TITLE
Updated to .NET 5 wherever possible

### PR DESCRIPTION
**Pull request by tomasherceg**
**Important: this PR is based on #872 (to prevent possible conflicts). Do not review & merge before merging new CLI first.**

Some projects were targetting .NET Core 2.0 or 3.0 which are out of support.

I've updated:
* ASP.NET Core samples to .NET Core 2.1 (lowest LTS version)
* ASP.NET Core latest samples to .NET 5.0
* all test projects to .NET 5.0

Versions of the framework itself remain as they are.

I've also fixed some warnings also, but we have still quite a lot of them (mostly related to nullable reference types).

### CODE REVIEW

**Comment by quigamdev** [comment]
This PR is ready for review.

File: src/DotVVM.Compiler/DotVVM.Compiler.csproj [hunk 1]
```diff
@@ -19,6 +19,9 @@ for some reason. What fun. -->
     <CheckEolTargetFramework>false</CheckEolTargetFramework>
     <RollForward>LatestMajor</RollForward>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net461'">
+    <RuntimeIdentifiers>win7-x86;win7-x64</RuntimeIdentifiers>
+  </PropertyGroup>
```
Comment (cafstep) [inline src/DotVVM.Compiler/DotVVM.Compiler.csproj:24]: Why is this needed? The Compiler binaries created upon the build of DotVVM.CommandLine are framework-dependent, not self-contained, so this setting shouldn't really do much, right?

File: src/DotVVM.Framework.Api.Swashbuckle.AspNetCore.Tests/DotVVM.Framework.Api.Swashbuckle.AspNetCore.Tests.csproj [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.App" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
-    <PackageReference Include="MSTest.TestAdapter" Version="2.1.1" />
-    <PackageReference Include="MSTest.TestFramework" Version="2.1.1" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.8.3" />
+    <PackageReference Include="MSTest.TestAdapter" Version="2.1.2" />
+    <PackageReference Include="MSTest.TestFramework" Version="2.1.2" />
```
Comment (cafstep) [inline src/DotVVM.Framework.Api.Swashbuckle.AspNetCore.Tests/DotVVM.Framework.Api.Swashbuckle.AspNetCore.Tests.csproj:12]: Since this PR is now almost 6 months old, should we update packages all around again?

File: src/DotVVM.Compiler/DotVVM.Compiler.csproj (hunk 1 above)
Comment (quigamdev) [reply src/DotVVM.Compiler/DotVVM.Compiler.csproj:24]: I think this was necessary for the preview version, but I am not sure whether this is still needed.

**Comment by quigamdev** [comment]
Fixed all issues. Should be good to go. @cafstep @tomasherceg

**Review by quigamdev — APPROVED** [review]

File: src/DotVVM.Samples.Common/ViewModels/ControlSamples/GridView/GridViewPagingSortingViewModel.cs [hunk 3]
```diff
@@ -81,7 +77,16 @@ public override Task PreRender()
 
         public void TestCommand()
         {
-            CustomersDataSet.SetSortExpression("BirthDate");
+            if (CustomersDataSet.SortingOptions.SortExpression == "BirthDate")
+            {
+                CustomersDataSet.SortingOptions.SortDescending = !CustomersDataSet.SortingOptions.SortDescending;
```
Comment (exyi) [inline src/DotVVM.Samples.Common/ViewModels/ControlSamples/GridView/GridViewPagingSortingViewModel.cs:82]:
```suggestion
                CustomersDataSet.SortingOptions.SortDescending ^= true;
```

;)

**Review by exyi — APPROVED** [review]
